pkg/apis/experimental/validation/validation.go

```diff
@@ -333,3 +345,130 @@ func ValidateJobSpecUpdate(oldSpec, spec experimental.JobSpec) errs.ValidationEr
 	}
 	return allErrs
 }
+
+// ValidateIngress tests if required fields in the Ingress are set.
+func ValidateIngress(ingress *experimental.Ingress) errs.ValidationErrorList {
+	allErrs := errs.ValidationErrorList{}
+	allErrs = append(allErrs, apivalidation.ValidateObjectMeta(&ingress.ObjectMeta, true, ValidateIngressName).Prefix("metadata")...)
+	allErrs = append(allErrs, ValidateIngressSpec(&ingress.Spec).Prefix("spec")...)
+
+	return allErrs
+}
+
+// ValidateIngressName validates that the given name can be used as an Ingress name.
+func ValidateIngressName(name string, prefix bool) (bool, string) {
+	return apivalidation.NameIsDNSSubdomain(name, prefix)
+}
+
+// ValidateIngressSpec tests if required fields in the IngressSpec are set.
+func ValidateIngressSpec(spec *experimental.IngressSpec) errs.ValidationErrorList {
+	allErrs := errs.ValidationErrorList{}
+	// TODO: Is a default backend mandatory?
+	if spec.Backend != nil {
```

Comment:
I think it's confusing to require a default like gce does. It forces you to take one from your paths and dump it there or create a new one, increasing the size of what should be a tiny resource. I was hoping (and this might be a stretch) to serve some sort of index page on apiserver and default it as the backend. 

In the case you mentioned curl will hang for a bit and then err out the browser will do whatever it does for bad url.